syntax = "proto3";

package image;

message Request {
    string image_data = 1;
    int32 req_id = 2;
}

message Response{
    string ack_data = 1;
    int32 req_id = 2;
    double recv_time = 3;
}

service GRPCImage {
    rpc ProcessImageSync (Request) returns (Response);
    rpc ProcessImageStreaming (stream Request) returns (stream Response);
}
